### PR TITLE
SDK-295 [Python SDK] Sorting support

### PR DESCRIPTION
**Pull request by RommeroInCountry**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- [SDK-295](https://incountry.atlassian.net/browse/SDK-295)

Code Quality (Before submitting PR)
------------
- [ ] Ensure code compiles and passes Sonar Lint.
- [ ] Do your best to ensure integration tests pass. If they don't then...
- [ ] Remove, ignore, or mark pending any brittle tests.
- [ ] Remove commented out code.

Target Release
--------------
- [ ] Please put the target release vehicle this needs to be bundled with.
- [ ] Did you increment the version? What is the new version?

### CODE REVIEW

File: incountry/storage.py [hunk 1]
```diff
@@ -233,6 +235,9 @@ def find(
             country: Country to search records in
             limit: Maximum amount of records to be returned. Max limit is 100. Defaults to 100.
             offset: Search offset. Should be non-negative int. Defaults to 0.
+            sort: Sort filter.
+                An array of single-key dicts allowing to sort result data, e.g:
+                [{"key1": "asc"}, {"key2": "desc"}]
             **filters: Various filters to tweak the search query.
                 Available String filter keys:
                 - profile_key, service_key1, service_key2, key1, ..., key10.
```
Comment (VladimirLevitskiy) [inline incountry/storage.py:243]: key10 -> key20

File: incountry/storage.py [hunk 2]
```diff
@@ -318,13 +327,17 @@ def find_one(
...
 
         Args:
             country: Country to search record in
             offset: Search offset. Should be non-negative int. Defaults to 0.
+            sort: Sort filter.
+                An array of single-key dicts allowing to sort result data, e.g:
+                [{"key1": "asc"}, {"key2": "desc"}]
             **filters: Various filters to tweak the search query.
                 Available String filter keys:
                 - profile_key, service_key1, service_key2, key1, ..., key10,
```
Comment (VladimirLevitskiy) [inline incountry/storage.py:343]: the same, key10 -> key20

File: incountry/storage.py [hunk 3]
```diff
@@ -233,6 +235,9 @@ def find(
             country: Country to search records in
             limit: Maximum amount of records to be returned. Max limit is 100. Defaults to 100.
             offset: Search offset. Should be non-negative int. Defaults to 0.
+            sort: Sort filter.
```
Comment (VladimirLevitskiy) [inline incountry/storage.py:238]: Please add a section to the README

**Review by VladimirLevitskiy — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -433,9 +434,32 @@ version={"$not": 1} # records with version not equal 1
 version={"$not": [1, 2]} # records with version equal to neither 1 or 2
 ```
 
+#### Sorting
+
+You can use `sort` parameter to sort the returned records by one or multiple keys.
+SDK currently supports sorting by `key1...key20`, `range_key1...range_key10` keys.
```
Comment (VladimirLevitskiy) [inline README.md:440]: What about date fields?

**Review by VladimirLevitskiy — CHANGES_REQUESTED** [review]

File: tests/unit/storage/test_enpoint_countries_endpoint.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
+import json
```
Comment (artemgolozubov) [inline tests/unit/storage/test_enpoint_countries_endpoint.py:1]: Pls fix a typo in filename

File: README.md [hunk 6]
```diff
@@ -433,9 +434,32 @@ version={"$not": 1} # records with version not equal 1
...
+
+# or you can use string values
+sort=[{"key1": "asc"}, {"range_key1": "desc"}]
+```
+
+---
+
+**WARNING**
+
+To use `sort` in find() call for string keys `key1...key20` you need to set  `hashSearchKeys` option to `false`.
```
Comment (artemgolozubov) [inline README.md:456]:
```suggestion
To use `sort` in find() call for string keys `key1...key20` you need to set  `hash_search_keys` option to `False`.
```

**Review by artemgolozubov — COMMENTED** [review]
Pls take a look at the comments below

File: README.md (hunk 4 above)
Comment (RommeroInCountry) [reply README.md:440]: Thanks for noticing

File: incountry/models/sort_filter.py [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+    @validator("*", pre=True)
+    def forbid_multiple_options(cls, value, values, config, field):
+        used_key = next(filter(lambda k: values.get(k, None), values.keys()), None)
+        if used_key:
+            raise ValueError(f"sort option may have only one field selected, already got '{used_key}'")
+        return value
+
+
+class SortFilter(BaseModel):
```
[the remark this answers is not in the document]
Comment (RommeroInCountry) [reply incountry/models/sort_filter.py:61]: my bad

**Review by artemgolozubov — APPROVED** [review]
LGTM

**Review by VladimirLevitskiy — APPROVED** [review]

**Pull request merged**